geomstats/learning/geometric_median.py

```diff
@@ -1,116 +1,135 @@
     lr : float
         Learning rate to be used for the algorithm.
         Optional, default : 1.0
-    init : array-like
+    init : array-like, shape={representation shape}
         Initialization to be used in the start.
         Optional, default : None
     print_every : int
         Print updated median after print_every iterations.
         Optional, default : None
+    epsilon : float
+        Tolerance for stopping the algorithm (distance between two successive
+        estimates).
+        Optional, default : gs.atol
+
+    Attributes
+    ----------
+    estimate_ : array-like, shape={representation shape}
+        If fit, geometric median.
 
     References
     ----------
-    .. [BJL2008]_ Bhatia, Jain, Lim. "Robust statistics on
-        Riemannian manifolds via the geometric median"
+    .. [FVJ2009] Fletcher PT, Venkatasubramanian S and Joshi S.
+        "The geometric median on Riemannian manifolds with application to
+        robust atlas estimation", NeuroImage, 2009
+        https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2735114/
     """
 
-    def __init__(self, metric, max_iter=100, lr=1.0, init=None, print_every=None):
+    def __init__(
+        self,
+        metric,
+        max_iter=100,
+        lr=1.0,
+        init=None,
+        print_every=None,
+        epsilon=gs.atol,
+    ):
         self.metric = metric
         self.max_iter = max_iter
         self.lr = lr
         self.init = init
         self.print_every = print_every
+        self.epsilon = epsilon
         self.estimate_ = None
 
     def _iterate_once(self, current_median, X, weights, lr):
-        """Compute a single iteration of Weiszfeld Algorithm.
+        """Compute a single iteration of Weiszfeld algorithm.
 
         Parameters
         ----------
         current_median : array-like, shape={representation shape}
-            current median.
-        X : array-like, shape=[..., {representation shape}]
-            data for which geometric has to be found.
-        weights : array-like, shape=[N]
-            weights for weighted sum.
+            Current median.
+        X : array-like, shape=[n_samples, {representation shape}]
+            Training input samples.
+        weights : array-like, shape=[n_samples,]
+            Weights for weighted sum.
         lr : float
-            learning rate for the current iteration.
+            Learning rate for the current iteration.
 
         Returns
         -------
-        updated_median: array-like, shape={representation shape}
-            updated median after single iteration.
+        updated_median : array-like, shape={representation shape}
+            Updated median after single iteration.
         """
-
-        def _scalarmul(scalar, array):
-            if gs.ndim(array) == 2:
-                return scalar[:, None] * array
-
-            return scalar[:, None, None] * array
-
         dists = self.metric.dist(current_median, X)
+        is_zero = dists == 0
```

Comment:
`is_not_zero = dists > gs.atol` instead? avoids negation and adds tolerance control, as it will never be zero.